### PR TITLE
Add-mip-30

### PR DESCRIPTION
**Pull request by dimsome**

### CODE REVIEW

File: MIPS/mip-30.md [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+## Abstract
+
+### Important comment upfront:
+
+- **_This MIP is NOT to discuss whether to sunset or not_**, this will be part of a future proposal. But in light of MIP 29, we need to prepare for the eventuality of the shutdown of the products and how it’s going to be handled on a smart contract level.
+- **_This MIP outlines HOW to sunset the products_** and to offer transparency around the guiding principles, methodology and smart contract changes/upgrades.
+- **_The actual decision to execute this will be done in a future proposal._** A future proposal will be made that is referenced in MIP 29 that will offer options for proposed continuation as well as to pursue the shutdown.
+- In the case that a proposal for Merge/Acquisition/Continuation is successful, this Proposal and the formal proposal resulting from it will be superseded by the winning proposal
+
+This document serves to discuss the guiding principles and methodology for sunsetting the products and smart contracts in accordance with [MIP 29](https://mips.mstable.org/MIPS/mip-29.html).
```
Comment (rafaelugolini) [inline MIPS/mip-30.md:25]: "sunsetting the products and smart contracts" or "sunsetting the products and their smart contracts"?

File: MIPS/mip-30.md [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+
+The first guiding principle for sunsetting the products and smart contracts is to allow user withdrawals at any time in the future and to never lock user funds. This means that any user will be able to access their funds at any point in the future, with no restrictions. It also means that users can be sure that their funds are safe, as they are not kept locked away in the contract.
```
Comment (rafaelugolini) [inline MIPS/mip-30.md:27]: same here

File: MIPS/mip-30.md [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+
+Outlined here is the high-level approach to the sunset of each product. Staking contracts are excluded here and will be further specified in another proposal outlining MTA procedure.
+
+### mAssets and Save
+
+- Upgrade the mAssets’s `migrateBassets` to support migrating bAssets from platforms back to the mAsset contract
+- Remove platform integration
+- Set swap and redeem fee to 0
+- Set new min-max weights (5% and 95%)
+- This applies to mUSD and mBTC on Ethereum mainnet and to mUSD on Polygon
```
Comment (rafaelugolini) [inline MIPS/mip-30.md:55]: should we add something `mBTC(deprecated)`

File: MIPS/mip-30.md (hunk 2 above)
Comment (dimsome) [reply MIPS/mip-30.md:27]: ok

File: MIPS/mip-30.md (hunk 3 above)
Comment (dimsome) [reply MIPS/mip-30.md:55]: I don't think so, because we never actually resolved anything here, never changed the mBTC smart contracts.

**Review by rafaelugolini — APPROVED** [review]

**Pull request merged**